es6-shim.js

```diff
@@ -2097,6 +2170,257 @@
     // Shim incomplete iterator implementations.
     addIterator(Object.getPrototypeOf((new globals.Map()).keys()));
     addIterator(Object.getPrototypeOf((new globals.Set()).keys()));
+
+    // Reflect
+    if (!globals.Reflect) {
+      var Reflect = {};
+
+      var wrapObjectFunction = function (func) {
+        return function (target) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          try {
+            func.apply(Object, arguments);
```

Comment:
Wrapping everything in try/catch makes things really slow - is this necessary for every case?